### PR TITLE
Test Twilio Phone Verfication Database Insertion

### PR DESCRIPTION
**Pull request by donald-mate**
This creates a test to ensure a mock JSON Twilio response can be saved as a valid PhoneNumberVerification entry. A second test was added that checks the input of the mock Twilio response to what is saved in the database.

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -137,6 +137,15 @@ class PhoneNumberVerification(db.Model):
     country_code = db.Column(db.String(10))
     national_format = db.Column(db.String(20))
     phone_number = db.Column(db.String(30))
+    def from_twilio_response(input):
+        return PhoneNumberVerification(
+            caller_name = input['caller_name']['caller_name'],
+            caller_type = input['caller_name']['caller_type'],
+            error_code = input['caller_name']['error_code'],
```
Comment (thebritican) [inline app/models.py:144]: I think we're going to need to account for `input['caller_name']` sometimes being `null`.

https://www.twilio.com/docs/lookup/api#lookups-caller-name

image

I think this is what you want. I'd add a test for when each of these fields is `null` to make sure we don't blow up when those values come back from the API

```suggestion
caller_info = input['caller_name'] or { 'caller_name': None, 'caller_type': None, 'error_code': None }

return PhoneNumberVerification(
    caller_name = caller_info['caller_name'],
    caller_type = caller_info['caller_type'],
    error_code = caller_info['error_code'],
```

File: tests/test_mock_twilio.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "caller_type": "CONSUMER",
+    "error_code": null
+  },
+  "carrier": null,
+  "country_code": "US",
+  "national_format": "(510) 867-5310",
+  "phone_number": "+15108675310",
+  "add_ons": null,
+  "url": "https://lookups.twilio.com/v1/PhoneNumbers/+15108675310"
+}''')
```
Comment (thebritican) [inline tests/test_mock_twilio.py:22]: can you try putting this in a separate file? we can add a directory `tests/fixtures/` and put this into `tests/fixtures/phone_number_with_caller_name.json`

Then, we can add a file `tests/fixtures/phone_number_missing_caller_name.json` for the test where `input['caller_name']` is `null`.

File: tests/test_mock_twilio.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  
+output = PhoneNumberVerification.from_twilio_response(input)
+
```
Comment (thebritican) [inline tests/test_mock_twilio.py:25]: to actually save this to the database, i would use `db.session`

```suggestion
db.session.add(output) # this adds your python object to the SQL transaction
db.session.commit() # this executes the transaction, actually running `INSERT` statements

# this queries the phone_number_verifications SQL table, grabbing the first entry
phone_number_entry = db.session.query(PhoneNumberVerification).first() 
```
 I would use `phone_number_entry` in your test instead of `output` Why? `output` is testing that JSON becomes a Python object. We want to test that the Python object can be inserted into the DB (SQL) and that when we ask for it with a query, it's returned from the db with all the correct data.

**Comment by thebritican** [comment]
This will resolve #5
